daemon/daemon.go

```diff
@@ -355,10 +344,35 @@ func NewDaemon(dp datapath.Datapath, iptablesManager rulesManager) (*Daemon, *en
 		datapath:          dp,
 		nodeDiscovery:     nodediscovery.NewNodeDiscovery(nodeMngr, mtuConfig),
 		iptablesManager:   iptablesManager,
-		endpointManager:   epMgr,
 	}
+
+	d.identityAllocator = cache.NewIdentityAllocatorManager(&d)
+	d.policy = policy.NewPolicyRepository(d.identityAllocator)
+
+	// Propagate identity allocator down to packages which themselves do not
+	// have types to which we can add an allocator member.
+	//
+	// TODO: convert these package level variables to types for easier unit
+	// testing in the future.
+	ipcache.IdentityAllocator = d.identityAllocator
+	proxy.Allocator = d.identityAllocator
+
+	d.endpointManager = endpointmanager.NewEndpointManager(&endpointsynchronizer.EndpointSynchronizer{
+		Allocator: d.identityAllocator,
+	})
+	d.endpointManager.InitMetrics()
```

Comment:
nit: Why does the daemon need to invoke `InitMetrics()`, shouldn't this detail be handled by `NewEndpointManager()`?